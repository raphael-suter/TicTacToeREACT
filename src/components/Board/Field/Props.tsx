export default interface Props {
  content: string;
  onClick: () => void;
};
