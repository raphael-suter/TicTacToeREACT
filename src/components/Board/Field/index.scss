@import "../../../styles/colors.scss";

.field {
  margin: 0.3rem;
  position: relative;

  &::before {
    content: "";
    display: block;
    padding-bottom: 100%;
  }

  button {
    position: absolute;
    top: 0;
    bottom: 0;
    width: 100%;
    font-size: 52px;
    background: $mediumColor;
    box-shadow: inset 0 0 2px 0 $shadow;
    border: none;
    cursor: pointer;
  }
}
