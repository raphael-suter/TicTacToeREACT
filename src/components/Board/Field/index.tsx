import React from 'react';
import './index.scss';
import Props from './Props';

export default ({ content, onClick }: Props) => (
  <div className='board-field'>
    <button onClick={onClick} aria-label='field'>{content}</button>
  </div>
);
