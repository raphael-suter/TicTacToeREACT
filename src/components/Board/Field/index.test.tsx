import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import Field from '.';

const setup = (onClick: () => void) => shallow(<Field content='X' onClick={onClick} />);

describe('Field', () => {
  test('Renders correctly.', () => {
    const wrapper = setup(null);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('Fires click events.', () => {
    const onClick = jasmine.createSpy();
    setup(onClick).childAt(0).simulate('click');

    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
