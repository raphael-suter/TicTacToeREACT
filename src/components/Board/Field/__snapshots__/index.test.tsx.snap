// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field Renders correctly. 1`] = `
<styled.div>
  <styled.button
    aria-label="field"
    onClick={null}
  >
    X
  </styled.button>
</styled.div>
`;
